### DISCUSSION TITLE
conflict with S3Servo

### DISCUSSION THREAD

**Discussion by Jeronimo59** [pschatzmann/arduino-audio-tools#1569] [Q&A] [answered]
Hello,

I am trying to combine PWM Audio playing and servo control using a [Xiao ESP32-S3 Sense](https://wiki.seeedstudio.com/xiao_esp32s3_getting_started/)

I managed to play nicely   .WAV files (@11025Hz) from SD card using PWMAudioOutput thanks to your library, but when I add a servo using [ESP32S3Servo library](https://github.com/SimGallery/ESP32S3servo.git), I get repetitive "clicks" on the speaker on top of sound, and servo does not work (it works when I disable arduino-audio-tools).
 
ESP32S3servo uses basic ESP32 PWM control channels (ledcSetup, ledcAttachPin, ledcWrite). I tried to use different channels, but with no success. Any suggestion on where the conflict could be and how to address it ?

Thanks,

**Comment by pschatzmann** [comment]
The AudioTools use the very same API. I am not sure if you can use these two libraries together because the might have different requirements in regard of the clock configuration.

In any case you will need to make sure that you use different channels then the ones that are already used. You can check this when you activate the log level info

**Reply by Jeronimo59** [reply]
Thanks for your prompt answer. 

I checked the logs (info or Debug) but could not find the designated PWM channel(s). Am I missing something ? Here are the logs .

[I] AudioTypes.h : 127 -  sample_rate: 8000 / channels: 1 / bits_per_sample: 16
[I] PWMAudioBase.h : 124 - decimation: 1
[I] PWMAudioBase.h : 133 - ->Allocating new buffer 4 * 1024 bytes
[I] Buffers.h : 372 - resize: 4096
[I] PWMAudioBase.h : 87 - samp[I] StreamCopy.h : 147 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops

**Reply by pschatzmann** [reply, marked as answer]
Maybe you need to add some wait loop at the beginning of your sketch (after Serial.begin()) to wait for Serial to be ready:
while(!Serial);

**Reply by Jeronimo59** [reply]
Indeed I found that using channel 2 or 3 - having found audio_tools used channel 0 - does the trick :-).
